site/source/docs/debugging/Sanitizers.rst

```diff
@@ -104,3 +104,196 @@ The minimal runtime is supported by Emscripten. To use it, pass the flag
+  WRITE of size 11 at 0x02965e5a thread T0
+      #0 0x15f0 in __asan_memset+0x15f0 (a.out.wasm+0x15f0)
+      #1 0xc46 in __original_main stack_buffer_overflow.c:5:3
+      #2 0xcbc in main+0xcbc (a.out.wasm+0xcbc)
+      #3 0x800019bc in Object.Module._main a.out.js:6588:32
+      #4 0x80001aeb in Object.callMain a.out.js:6891:30
+      #5 0x80001b25 in doRun a.out.js:6949:60
+      #6 0x80001b33 in run a.out.js:6963:5
+      #7 0x80001ad6 in runCaller a.out.js:6870:29
+
+  Address 0x02965e5a is located in stack of thread T0 at offset 26 in frame
+      #0 0x11  (a.out.wasm+0x11)
+
+    This frame has 1 object(s):
+      [16, 26) 'x' (line 4) <== Memory access at offset 26 overflows this variable
+  HINT: this may be a false positive if your program uses some custom stack unwind mechanism, swapcontext or vfork
+        (longjmp and C++ exceptions *are* supported)
+  SUMMARY: AddressSanitizer: stack-buffer-overflow (a.out.wasm+0x15ef)
+  ...
+
+Use After Free
+^^^^^^^^^^^^^^
+
+Consider ``use_after_free.cpp``:
+
+.. code-block:: cpp
+
+  int main() {
+    int *array = new int[100];
+    delete [] array;
+    return array[0];
+  }
+
+.. code-block:: console
+
+  $ em++ -g4 -fsanitize=address -s ALLOW_MEMORY_GROWTH use_after_free.cpp
+  $ node a.out.js
+  =================================================================
+  ==42==ERROR: AddressSanitizer: heap-use-after-free on address 0x03203e40 at pc 0x00000c1b bp 0x02965e70 sp 0x02965e7c
+  READ of size 4 at 0x03203e40 thread T0
+      #0 0xc1b in __original_main use_after_free.cpp:4:10
+      #1 0xc48 in main+0xc48 (a.out.wasm+0xc48)
+
+  0x03203e40 is located 0 bytes inside of 400-byte region [0x03203e40,0x03203fd0)
+  freed by thread T0 here:
+      #0 0x5fe8 in operator delete[](void*)+0x5fe8 (a.out.wasm+0x5fe8)
+      #1 0xb76 in __original_main use_after_free.cpp:3:3
+      #2 0xc48 in main+0xc48 (a.out.wasm+0xc48)
+      #3 0x800019b5 in Object.Module._main a.out.js:6581:32
+      #4 0x80001ade in Object.callMain a.out.js:6878:30
+      #5 0x80001b18 in doRun a.out.js:6936:60
+      #6 0x80001b26 in run a.out.js:6950:5
+      #7 0x80001ac9 in runCaller a.out.js:6857:29
+
+  previously allocated by thread T0 here:
+      #0 0x5db4 in operator new[](unsigned long)+0x5db4 (a.out.wasm+0x5db4)
+      #1 0xb41 in __original_main use_after_free.cpp:2:16
+      #2 0xc48 in main+0xc48 (a.out.wasm+0xc48)
+      #3 0x800019b5 in Object.Module._main a.out.js:6581:32
+      #4 0x80001ade in Object.callMain a.out.js:6878:30
+      #5 0x80001b18 in doRun a.out.js:6936:60
+      #6 0x80001b26 in run a.out.js:6950:5
+      #7 0x80001ac9 in runCaller a.out.js:6857:29
+
+  SUMMARY: AddressSanitizer: heap-use-after-free (a.out.wasm+0xc1a)
+  ...
+
+Memory Leaks
+^^^^^^^^^^^^
+
+Consider ``leak.cpp``:
+
+.. code-block:: cpp
+
+  int main() {
+    new int[10];
+  }
+
+Note that since leak checks take place at program exit, you must use
+``-s EXIT_RUNTIME``, or invoke ``__lsan::DoLeakCheck`` manually.
```

Comment:
Is there a macro users can use to determine whether ASan is enabled so they can conditionally call `__lsan::DoLeakCheck` only for ASan builds?